### ISSUE TITLE
feature_extractor/for_image_data model loading probelm

### ISSUE THREAD

**Issue by openyuanmu** [jindongwang/transferlearning#137]
**Before you open an issue**
If it's code running error, maybe you want to check the python or pytorch version before submitting an issue.
pytorch=1.3.1

**Describe the bug**
```
model = models.Network(base_net=args.model_name,
                               n_class=args.num_class)
model.load_state_dict(torch.load(model_path))
```
Error: _IncompatibleKeys(...)

**Solution**
```
ckpt_path = os.path.join("weights", dataset, "{}.pth".format(domain))
if not torch.cuda.is_available():
	ckpt = torch.load(ckpt_path, map_location=torch.device('cpu'))
else:
	ckpt = torch.load(ckpt_path)

new_ckpt = {}
for k,v in ckpt.items():
	if k == "fc.weight":
		new_ckpt["classifier_layer.weight"] = v
	elif k == "fc.bias":
		new_ckpt["classifier_layer.bias"] = v
	else:
		new_ckpt["base_network."+k] = v
```
above code can load the weight correctly.

**Comment by openyuanmu** [comment]
Also, I think the first performed transform should be `transforms.Resize((224,224))` instead of `transforms.Resize(224)`.

Below is my script to use the pretrained resnet50 model, hope it will be helpful:
```
"""
run_resnet50

yuanmu

run resnet50 models on office_home/image_clef dataset
and save output as npy file
"""
import os
import models
import torch
from PIL import Image
from torchvision import transforms
from torch.nn import functional as F
import numpy as np

# ===================================
# datasets: office_home / image_clef
dataset = "office_home"
# domains:
# 	office_home: ["art", "clipart", "product", "real_world"]
# 	image_clef: ["c", "i", "p"]
domain = "art"
# splits: train / val
split = "train"
# ===================================
output_path = "output/resnet50_output_on_{}_{}_{}.npy".format(dataset, domain, split)

if dataset == "office_home":
	root_dir = "/home/yuanmu/workspace/dataset/OfficeHome/"
	n_class = 65
elif dataset == "image_clef":
	root_dir = "/home/yuanmu/workspace/dataset/image_CLEF/"
	n_class = 12

list_path = os.path.join(root_dir, "{}_{}_{}.txt".format(dataset, domain, split))

# num_class:
# 	office_home: 65
# 	image_clef: 12
model = models.Network(base_net='resnet50', n_class=n_class)
ckpt_path = os.path.join("weights", dataset, "{}.pth".format(domain))
if not torch.cuda.is_available():
	ckpt = torch.load(ckpt_path, map_location=torch.device('cpu'))
else:
	ckpt = torch.load(ckpt_path)

new_ckpt = {}
for k,v in ckpt.items():
	if k == "fc.weight":
		new_ckpt["classifier_layer.weight"] = v
	elif k == "fc.bias":
		new_ckpt["classifier_layer.bias"] = v
	else:
		new_ckpt["base_network."+k] = v

model.load_state_dict(new_ckpt)
model.eval()
if torch.cuda.is_available():
	model = model.cuda()

transform_test = transforms.Compose([
	transforms.Resize((224,224)),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406],std=[0.229, 0.224, 0.225])])


output = []
with open(list_path, "r") as fin:
	count = 0
	for line in fin.readlines():
		img_path, catid = line.strip().split(' ')
		img = Image.open(os.path.join(root_dir, img_path))
		img = transform_test(img).unsqueeze(0)
		if torch.cuda.is_available():
			img = img.cuda()

		logit = model.forward(img)
		h_x = F.softmax(logit, 1).data.squeeze()
		output.append(h_x.numpy())
		count += 1
		if count % 20 == 0:
			print("{} processed.".format(count))

np.save(open(output_path, "wb"), output)
print("DONE.")
```

**Comment by openyuanmu** [comment]
**Update for grayscale image input.**
There are grayscale images in image_clef, so the transform should be updated as follows:

transform_test = transforms.Compose([
	transforms.Resize((224,224)),
    transforms.ToTensor()])
trans_norm = transforms.Normalize(mean=[0.485, 0.456, 0.406],std=[0.229, 0.224, 0.225])


```
....
output = []
with open(list_path, "r") as fin:
	count = 0
	for line in fin.readlines():
		img_path, catid = line.strip().split(' ')
		# print(img_path)
		img = Image.open(os.path.join(root_dir, img_path))
		# img = transform_test(img).unsqueeze(0)
		img = transform_test(img)
		# for grayscale images
		if img.shape[0] == 1:
			img = img.repeat(3,1,1)
		img = trans_norm(img).unsqueeze(0)

		if torch.cuda.is_available():
			img = img.cuda()
		
		# print(img.shape+'\n')
		logit = model.forward(img)
		h_x = F.softmax(logit, 1).data.squeeze()
		output.append(h_x.numpy())
		count += 1
		if count % 20 == 0:
			print("{} processed.".format(count))
....
```

**Comment by jindongwang** [comment]
Thanks for pointing out the issues. If you want to collaborate on these codes, I suggest you open a pull request where you can upload your code. Then I can merge them.
